.travis_suppressions

```diff
@@ -6,6 +6,8 @@ noValidConfiguration
 shadowFunction
 functionConst
 functionStatic
+bitwiseOnBoolean:lib/checkcondition.cpp
```

Comment:
I suggest that you suppress this in all files.

Reply:
Instead of suppressing this, can we not warn when `isConstExpression(tok->astOperand1())` is false?

Reply:
Updated to suppress in all files.

> We should check if the LHS or RHS has side effects.

My concern is that this would skip cases where this check is most useful. If the RHS has side effects and the bitwise op is unintentional, then the check as-is reveals a potential bug. That may be fairly rare, but it's hard to be sure of the programmer's intent.

Reply:
> We should check if the LHS or RHS has side effects.
> Instead of suppressing this, can we not warn when isConstExpression(tok->astOperand1()) is false?

I did not want to have this checker personally.. so I am not 100% sure. But I think those that wanted to have this checker wanted it to warn always for boolean operators.

I think we should keep these warnings.. but we can use a different ID if we see some special case.. however I am thinking that "separation" can be done separately. This PR is about making the checker work as it was intended to start with as far as I see.

If the logic behavior might be changed if `|` is replaced by `||` I think it would be best to mention this in the warning message.. amd possible use a different ID.. but I think that can be done in a separate PR also.